### PR TITLE
fix(Host): RHINENG-1891 sort hosts with several policies correctly

### PR DESCRIPTION
**Pull request by marleystipich2**
## Secure Coding Practices Checklist GitHub Link
- https://github.com/RedHatInsights/secure-coding-checklist

## Secure Coding Checklist
- [ ] Input Validation
- [ ] Output Encoding
- [ ] Authentication and Password Management
- [ ] Session Management
- [ ] Access Control
- [ ] Cryptographic Practices
- [ ] Error Handling and Logging
- [ ] Data Protection
- [ ] Communication Security
- [ ] System Configuration
- [ ] Database Security
- [ ] File Management
- [ ] Memory Management
- [x] General Coding Practices

### CODE REVIEW

File: app/models/host.rb [hunk 1]
```diff
@@ -99,8 +99,13 @@ class Host < ApplicationRecord
     )
   }
 
-  scope :joins_test_result_profiles, lambda {
-    left_outer_joins(:test_result_profiles)
+  scope :joins_test_result_profiles, lambda { |profile = nil|
+    profile ||= RequestStore.store['scoped_search_context_profiles']
+    profile_ids = profile&.pluck(:id) || []
```
Comment (vkrizan) [inline app/models/host.rb:104]: Can this be done by doing a separate query for ids?

Comment (marleystipich2) [reply app/models/host.rb:104]: Hmm I'm not sure what you mean by this.. Did you have something in mind?

Comment (vkrizan) [reply app/models/host.rb:104]: Here we're retrieving ids from the database and then passing it into another database query. Can we have the database deal with it internally?

Comment (vkrizan) [reply app/models/host.rb:104]: Sorry, I had a typo there, I meant "without doing a separate query".

Comment (marleystipich2) [reply app/models/host.rb:104]: Ah right I see what you mean now. Yeah, I can think of something.

**Comment by skateman** [comment]
@vkrizan are you okay with this?

**Review by romanblanco — APPROVED** [review]
LGTM :+1:

**Pull request merged**